### DISCUSSION TITLE
Removing duplicate tags

### DISCUSSION THREAD

**Discussion by morning-paradigm** [Difegue/LANraragi#775] [Q&A]
I got duplicate timestamp tags for some galleries after running the eze plugin.
Is there any way to get rid of duplicate tags?

I know I can batch remove the `timestamp` tags altogether, but that would require me to run the plugin again and it would take quite some time considering it fetches the data from E-H.

**Comment by Difegue** [comment]
Duplicate tags technically shouldn't happen (it's checked when inserting tags), but if there's a small difference in the text (`timestamp: a` vs `timestamp:a` for instance) you could get duplicates..  

I don't have a good built-in solution for removing dupe tags for now, although that's something that could be handled by a script plugin.

**Reply by morning-paradigm** [reply]
Alright, I've found out the timestamps fetched by E-Hentai plugin and eze plugin are different. In some cases it's off by 9 seconds, in some 51. I didn't notice it in the library since the dates are still the same.

This is weird as I thought the eze plugin fetched timestamps using the E-H plugin, but it actually uses `date_uploaded` from the JSON file generated with the eze (or rather "x") userscript. It still doesn't explain why the dates are off by a random amount however.

**Edit:** looks like it's an issue with the way `date_uploaded` is generated by the userscript. It ignores seconds, so that 16711146**51** becomes 16711146**00**
